docs/website/docs/technologies/mlir.md

```diff
@@ -0,0 +1,34 @@
+
+# MLIR
+
+[MLIR](https://mlir.llvm.org/) is the compiler framework that IREE is built
+around. Beyond the tooling this includes a set of common Dialects and
+transforms that IREE uses for it's code generation.
+
+Any required changes to MLIR should be upstreamed by following the MLIR
+contribution [guide](https://mlir.llvm.org/getting_started/Contributing/).
+IREE periodically syncs to MLIR's HEAD so there may be a noticeable delay
+between contributing to MLIR and having the feature available at IREE head.
```

Comment:
Some of these details might make more sense to include in one of these places instead of on the website:
* https://github.com/google/iree/blob/main/CONTRIBUTING.md
* https://github.com/google/iree/blob/main/docs/developers/developing_iree/contributor_tips.md
* https://github.com/google/iree/blob/main/docs/developers/developing_iree/repository_management.md (which is outdated)

In particular, I don't think we need to walk people through the "this might take a while" and "IREE HEAD vs MLIR HEAD" steps. (some portion of people reading these docs may not know what git is, and I don't think we need to explain all those specifics here)

Reply:
Removed the upstreaming information to prioritize on technologies.